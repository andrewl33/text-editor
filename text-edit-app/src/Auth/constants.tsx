// dashboard
// add file to account
// remove file from account
// add collection to account
// remove collection from account

// auth
// create account
// log in
// log out
export const CREATE_ACCOUNT_REQUEST = "CREATE_ACCOUNT_REQUEST";
export type CREATE_ACCOUNT_REQUEST = typeof CREATE_ACCOUNT_REQUEST;

export const CREATE_ACCOUNT_SUCCESS = "CREATE_ACCOUNT_SUCCESS";
export type CREATE_ACCOUNT_SUCCESS = typeof CREATE_ACCOUNT_SUCCESS;

export const CREATE_ACCOUNT_FAILURE = "CREATE_ACCOUNT_FAILURE";
export type CREATE_ACCOUNT_FAILURE = typeof CREATE_ACCOUNT_FAILURE;

export const LOG_IN_REQUEST = "LOG_IN_REQUEST";
export type LOG_IN_REQUEST = typeof LOG_IN_REQUEST;

export const LOG_IN_SUCCESS = "LOG_IN_SUCCESS";
export type LOG_IN_SUCCESS = typeof LOG_IN_SUCCESS;

export const LOG_IN_FAILURE = "LOG_IN_FAILURE";
export type LOG_IN_FAILURE = typeof LOG_IN_FAILURE;

export const LOG_OUT_REQUEST = "LOG_OUT_REQUEST";
export type LOG_OUT_REQUEST = typeof LOG_OUT_REQUEST;

export const LOG_OUT_SUCCESS = "LOG_OUT_SUCCESS";
export type LOG_OUT_SUCCESS = typeof LOG_OUT_SUCCESS;

export const LOG_OUT_FAILURE = "LOG_OUT_FAILURE";
export type LOG_OUT_FAILURE = typeof LOG_OUT_FAILURE;

export const LOG_IN_PROMPT = "LOG_IN_PROMPT";
export type LOG_IN_PROMPT = typeof LOG_IN_PROMPT;

export const CLOSE_PROMPT = "CLOSE_PROMPT";
export type CLOSE_PROMPT = typeof CLOSE_PROMPT;

export const UPDATE_TOKEN = "UPDATE_TOKEN";
export type UPDATE_TOKEN = typeof UPDATE_TOKEN;

export const GET_DASHBOARD_REQUEST = "GET_DASHBOARD_REQUEST";
export type GET_DASHBOARD_REQUEST = typeof GET_DASHBOARD_REQUEST;

export const GET_DASHBOARD_SUCCESS = "GET_DASHBOARD_SUCCESS";
export type GET_DASHBOARD_SUCCESS = typeof GET_DASHBOARD_SUCCESS;

export const GET_DASHBOARD_FAILURE = "GET_DASHBOARD_FAILURE";
export type GET_DASHBOARD_FAILURE = typeof GET_DASHBOARD_FAILURE;
