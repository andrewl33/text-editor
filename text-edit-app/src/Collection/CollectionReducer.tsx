import { CollectionAction } from './CollectionAction';
import {
  NEW_COLLECTION_REQUEST, NEW_COLLECTION_SUCCESS, NEW_COLLECTION_FAILURE,
  GET_COLLECTION_REQUEST, GET_COLLECTION_SUCCESS, GET_COLLECTION_FAILURE,
  AUTH_COLLECTION_REQUEST, AUTH_COLLECTION_SUCCESS, AUTH_COLLECTION_FAILURE,
  LOCK_COLLECTION_REQUEST, LOCK_COLLECTION_SUCCESS, LOCK_COLLECTION_FAILURE,
  ADD_FILE_REQUEST, ADD_FILE_SUCCESS, ADD_FILE_FAILURE,
  REMOVE_FILE_REQUEST, REMOVE_FILE_SUCCESS, REMOVE_FILE_FAILURE,
} from './constants';
import {
  SHARE_LINK, CLOSE_ALERT
} from '../constants';
import { CollectionStoreState } from '../types';

export const initialState: CollectionStoreState = {
  items: null,
  openAlert: false,
  alertMessage: '',
  isLocked: false
}

export const collectionReducer = (state: CollectionStoreState = initialState, action: CollectionAction) => {

  switch(action.type) {

    case NEW_COLLECTION_REQUEST:
      return {...state};
    case NEW_COLLECTION_SUCCESS:
      return {...state};
    case NEW_COLLECTION_FAILURE:
      return {...state};  

    case GET_COLLECTION_REQUEST:
      return {...state};
    case GET_COLLECTION_SUCCESS:
      return {...state, items: action.payload && action.payload.items};  
    case GET_COLLECTION_FAILURE:
      return {...state};  

    case AUTH_COLLECTION_REQUEST:
      return {...state};  
    case AUTH_COLLECTION_SUCCESS:
      return {...state};  
    case AUTH_COLLECTION_FAILURE:
      return {...state};  

    case LOCK_COLLECTION_REQUEST:
      return {...state};  
    case LOCK_COLLECTION_SUCCESS:
      return {...state, isLocked: action.payload};      
    case LOCK_COLLECTION_FAILURE:
      return {...state};  

    case ADD_FILE_REQUEST:
    case ADD_FILE_SUCCESS:
    case ADD_FILE_FAILURE:

    case REMOVE_FILE_REQUEST:
    case REMOVE_FILE_SUCCESS:
    case REMOVE_FILE_FAILURE:

    case SHARE_LINK:
      return {...state, openAlert: true, alertMessage: 'Copied to clipboard'};
    case CLOSE_ALERT:
      return {...state, openAlert: false};
    
    default:
      return state;

  }

}