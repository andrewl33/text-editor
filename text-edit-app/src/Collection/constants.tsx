export const NEW_COLLECTION_REQUEST = "NEW_COLLECTION_REQUEST";
export type NEW_COLLECTION_REQUEST = typeof NEW_COLLECTION_REQUEST;

export const NEW_COLLECTION_SUCCESS = "NEW_COLLECTION_SUCCESS";
export type NEW_COLLECTION_SUCCESS = typeof NEW_COLLECTION_SUCCESS;

export const NEW_COLLECTION_FAILURE = "NEW_COLLECTION_FAILURE";
export type NEW_COLLECTION_FAILURE = typeof NEW_COLLECTION_FAILURE;

export const GET_COLLECTION_REQUEST = "GET_COLLECTION_REQUEST";
export type GET_COLLECTION_REQUEST = typeof GET_COLLECTION_REQUEST;

export const GET_COLLECTION_SUCCESS = "GET_COLLECTION_SUCCESS";
export type GET_COLLECTION_SUCCESS = typeof GET_COLLECTION_SUCCESS;

export const GET_COLLECTION_FAILURE = "GET_COLLECTION_FAILURE";
export type GET_COLLECTION_FAILURE = typeof GET_COLLECTION_FAILURE;

export const GET_COLLECTION_AUTH = "GET_COLLECTION_AUTH";
export type GET_COLLECTION_AUTH = typeof GET_COLLECTION_AUTH;

export const AUTH_COLLECTION_REQUEST = "AUTH_COLLECTION_REQUEST";
export type AUTH_COLLECTION_REQUEST = typeof AUTH_COLLECTION_REQUEST;

export const AUTH_COLLECTION_SUCCESS = "AUTH_COLLECTION_SUCCESS";
export type AUTH_COLLECTION_SUCCESS = typeof AUTH_COLLECTION_SUCCESS;

export const AUTH_COLLECTION_FAILURE = "AUTH_COLLECTION_FAILURE";
export type AUTH_COLLECTION_FAILURE = typeof AUTH_COLLECTION_FAILURE;

export const LOCK_COLLECTION_REQUEST = "LOCK_COLLECTION_REQUEST";
export type LOCK_COLLECTION_REQUEST = typeof LOCK_COLLECTION_REQUEST;

export const LOCK_COLLECTION_SUCCESS = "LOCK_COLLECTION_SUCCESS";
export type LOCK_COLLECTION_SUCCESS = typeof LOCK_COLLECTION_SUCCESS;

export const LOCK_COLLECTION_FAILURE = "LOCK_COLLECTION_FAILURE";
export type LOCK_COLLECTION_FAILURE = typeof LOCK_COLLECTION_FAILURE;

export const ADD_FILE_TO_COLLECTION_REQUEST = "ADD_FILE_TO_COLLECTION_REQUEST";
export type ADD_FILE_TO_COLLECTION_REQUEST = typeof ADD_FILE_TO_COLLECTION_REQUEST;

export const ADD_FILE_TO_COLLECTION_SUCCESS = "ADD_FILE_TO_COLLECTION_SUCCESS";
export type ADD_FILE_TO_COLLECTION_SUCCESS = typeof ADD_FILE_TO_COLLECTION_SUCCESS;

export const ADD_FILE_TO_COLLECTION_FAILURE = "ADD_FILE_TO_COLLECTION_FAILURE";
export type ADD_FILE_TO_COLLECTION_FAILURE = typeof ADD_FILE_TO_COLLECTION_FAILURE;

export const REMOVE_FILE_FROM_COLLECTION_REQUEST =
  "REMOVE_FILE_FROM_COLLECTION_REQUEST";
export type REMOVE_FILE_FROM_COLLECTION_REQUEST = typeof REMOVE_FILE_FROM_COLLECTION_REQUEST;

export const REMOVE_FILE_FROM_COLLECTION_SUCCESS =
  "REMOVE_FILE_FROM_COLLECTION_SUCCESS";
export type REMOVE_FILE_FROM_COLLECTION_SUCCESS = typeof REMOVE_FILE_FROM_COLLECTION_SUCCESS;

export const REMOVE_FILE_FROM_COLLECTION_FAILURE =
  "REMOVE_FILE_FROM_COLLECTION_FAILURE";
export type REMOVE_FILE_FROM_COLLECTION_FAILURE = typeof REMOVE_FILE_FROM_COLLECTION_FAILURE;

export const CHANGE_COLLECTION_NAME_REQUEST = "CHANGE_COLLECTION_NAME_REQUEST";
export type CHANGE_COLLECTION_NAME_REQUEST = typeof CHANGE_COLLECTION_NAME_REQUEST;

export const CHANGE_COLLECTION_NAME_SUCCESS = "CHANGE_COLLECTION_NAME_SUCCESS";
export type CHANGE_COLLECTION_NAME_SUCCESS = typeof CHANGE_COLLECTION_NAME_SUCCESS;

export const CHANGE_COLLECTION_NAME_FAILURE = "CHANGE_COLLECTION_NAME_FAILURE";
export type CHANGE_COLLECTION_NAME_FAILURE = typeof CHANGE_COLLECTION_NAME_FAILURE;

export const ADD_USER_TO_COLLECTION_REQUEST = "ADD_USER_TO_COLLECTION_REQUEST";
export type ADD_USER_TO_COLLECTION_REQUEST = typeof ADD_USER_TO_COLLECTION_REQUEST;

export const ADD_USER_TO_COLLECTION_SUCCESS = "ADD_USER_TO_COLLECTION_SUCCESS";
export type ADD_USER_TO_COLLECTION_SUCCESS = typeof ADD_USER_TO_COLLECTION_SUCCESS;

export const ADD_USER_TO_COLLECTION_FAILURE = "ADD_USER_TO_COLLECTION_FAILURE";
export type ADD_USER_TO_COLLECTION_FAILURE = typeof ADD_USER_TO_COLLECTION_FAILURE;

export const REMOVE_USER_FROM_COLLECTION_REQUEST =
  "REMOVE_USER_FROM_COLLECTION_REQUEST";
export type REMOVE_USER_FROM_COLLECTION_REQUEST = typeof REMOVE_USER_FROM_COLLECTION_REQUEST;

export const REMOVE_USER_FROM_COLLECTION_SUCCESS =
  "REMOVE_USER_FROM_COLLECTION_SUCCESS";
export type REMOVE_USER_FROM_COLLECTION_SUCCESS = typeof REMOVE_USER_FROM_COLLECTION_SUCCESS;

export const REMOVE_USER_FROM_COLLECTION_FAILURE =
  "REMOVE_USER_FROM_COLLECTION_FAILURE";
export type REMOVE_USER_FROM_COLLECTION_FAILURE = typeof REMOVE_USER_FROM_COLLECTION_FAILURE;
