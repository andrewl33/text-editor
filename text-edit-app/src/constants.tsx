export const NEW_TEXT = 'NEW_TEXT';
export type NEW_TEXT = typeof NEW_TEXT;

export const LOCK_TEXT = 'LOCK_TEXT';
export type LOCK_TEXT = typeof LOCK_TEXT;

export const SHARE_LINK = 'SHARE_LINK';
export type SHARE_LINK = typeof SHARE_LINK;