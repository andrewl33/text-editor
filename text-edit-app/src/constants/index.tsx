// header
export const NEW_TEXT = 'NEW_TEXT';
export type NEW_TEXT = typeof NEW_TEXT;

export const LOCK_TEXT = 'LOCK_TEXT';
export type LOCK_TEXT = typeof LOCK_TEXT;

export const SHARE_LINK = 'SHARE_LINK';
export type SHARE_LINK = typeof SHARE_LINK;

// editor
export const UPDATE_CODE = 'UPDATE_CODE';
export type UPDATE_CODE = typeof UPDATE_CODE;

export const CHANGED_CODE = 'CHANGED_CODE';
export type CHANGED_CODE = typeof CHANGED_CODE;