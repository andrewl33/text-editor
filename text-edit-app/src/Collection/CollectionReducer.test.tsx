import * as constants from "../constants";
import { collectionReducer } from "./CollectionReducer";

describe("CollectionReducer", () => {
  test("it should return initial state", () => {
    expect(collectionReducer(undefined, {} as any)).toMatchSnapshot();
  });

  describe("newCollection", () => {
    test("it should handle NEW_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, { type: constants.NEW_COLLECTION_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle NEW_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.NEW_COLLECTION_SUCCESS,
          payload: {
            createDate: "1-1-2001"
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle NEW_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, { type: constants.NEW_COLLECTION_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("getCollection", () => {
    test("it should handle GET_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, { type: constants.GET_COLLECTION_REQUEST })
      ).toMatchSnapshot();
    });

    test("it should handle GET_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.GET_COLLECTION_SUCCESS,
          payload: {
            success: true,
            items: [],
            createDate: "1-1-2001",
            name: "",
            isLocked: false,
            users: []
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle GET_COLLECTION_AUTH", () => {
      expect(
        collectionReducer(undefined, { type: constants.GET_COLLECTION_AUTH })
      ).toMatchSnapshot();
    });

    test("it should handle GET_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, { type: constants.GET_COLLECTION_FAILURE })
      ).toMatchSnapshot();
    });
  });

  describe("authCollection", () => {
    test("it should handle AUTH_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.AUTH_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle AUTH_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.AUTH_COLLECTION_SUCCESS
        })
      ).toMatchSnapshot();
    });

    test("it should handle AUTH_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.AUTH_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });
  describe("lockCollection", () => {
    test("it should handle LOCK_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.LOCK_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle LOCK_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.LOCK_COLLECTION_SUCCESS
        })
      ).toMatchSnapshot();
    });

    test("it should handle LOCK_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.LOCK_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  describe("addUserToCollection", () => {
    test("it should handle ADD_USER_TO_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_USER_TO_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_USER_TO_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_USER_TO_COLLECTION_SUCCESS,
          payload: { success: true, accountName: "newAccount" }
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_USER_TO_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_USER_TO_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  describe("removeUserFromCollection", () => {
    test("it should handle REMOVE_USER_FROM_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_USER_FROM_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_USER_FROM_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_USER_FROM_COLLECTION_SUCCESS,
          payload: { accountName: "removedName", success: true }
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_USER_FROM_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_USER_FROM_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  describe("addFileToCollection", () => {
    test("it should handle ADD_FILE_TO_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_FILE_TO_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_FILE_TO_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_FILE_TO_COLLECTION_SUCCESS,
          payload: {
            success: true,
            newFileItem: { id: "1", name: "test", date: "1-1-2001", tags: [] }
          }
        })
      ).toMatchSnapshot();
    });

    test("it should handle ADD_FILE_TO_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.ADD_FILE_TO_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  describe("removeFileFromCollection", () => {
    test("it should handle REMOVE_FILE_FROM_COLLECTION_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_FILE_FROM_COLLECTION_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_FILE_FROM_COLLECTION_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_FILE_FROM_COLLECTION_SUCCESS,
          payload: { success: true, fileId: "10" }
        })
      ).toMatchSnapshot();
    });

    test("it should handle REMOVE_FILE_FROM_COLLECTION_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.REMOVE_FILE_FROM_COLLECTION_FAILURE
        })
      ).toMatchSnapshot();
    });
  });

  describe("changeCollectionName", () => {
    test("it should handle CHANGE_COLLECTION_NAME_REQUEST", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.CHANGE_COLLECTION_NAME_REQUEST
        })
      ).toMatchSnapshot();
    });

    test("it should handle CHANGE_COLLECTION_NAME_SUCCESS", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.CHANGE_COLLECTION_NAME_SUCCESS,
          payload: { success: true, name: "newName" }
        })
      ).toMatchSnapshot();
    });

    test("it should handle CHANGE_COLLECTION_NAME_FAILURE", () => {
      expect(
        collectionReducer(undefined, {
          type: constants.CHANGE_COLLECTION_NAME_FAILURE
        })
      ).toMatchSnapshot();
    });
  });
});
