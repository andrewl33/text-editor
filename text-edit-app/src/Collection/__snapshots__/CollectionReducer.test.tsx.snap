// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CollectionReducer addFileToCollection it should handle ADD_FILE_TO_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer addFileToCollection it should handle ADD_FILE_TO_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer addFileToCollection it should handle ADD_FILE_TO_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [
    Object {
      "date": "1-1-2001",
      "id": "1",
      "name": "test",
      "tags": Array [],
    },
  ],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer addUserToCollection it should handle ADD_USER_TO_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer addUserToCollection it should handle ADD_USER_TO_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer addUserToCollection it should handle ADD_USER_TO_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [
    "newAccount",
  ],
}
`;

exports[`CollectionReducer authCollection it should handle AUTH_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "Authentication Failed",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": true,
  "users": Array [],
}
`;

exports[`CollectionReducer authCollection it should handle AUTH_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "Authenticating",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": true,
  "users": Array [],
}
`;

exports[`CollectionReducer authCollection it should handle AUTH_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer changeCollectionName it should handle CHANGE_COLLECTION_NAME_FAILURE 1`] = `
Object {
  "alertMessage": "Could not connect to DB",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": true,
  "users": Array [],
}
`;

exports[`CollectionReducer changeCollectionName it should handle CHANGE_COLLECTION_NAME_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer changeCollectionName it should handle CHANGE_COLLECTION_NAME_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "newName",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer getCollection it should handle GET_COLLECTION_AUTH 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": true,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "prompt": "Private Collection",
  "users": Array [],
}
`;

exports[`CollectionReducer getCollection it should handle GET_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer getCollection it should handle GET_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer getCollection it should handle GET_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "1-1-2001",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer it should return initial state 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer lockCollection it should handle LOCK_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "Count not connect to DB",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": true,
  "users": Array [],
}
`;

exports[`CollectionReducer lockCollection it should handle LOCK_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer lockCollection it should handle LOCK_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": undefined,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer newCollection it should handle NEW_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer newCollection it should handle NEW_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer newCollection it should handle NEW_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "1-1-2001",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeFileFromCollection it should handle REMOVE_FILE_FROM_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeFileFromCollection it should handle REMOVE_FILE_FROM_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeFileFromCollection it should handle REMOVE_FILE_FROM_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeUserFromCollection it should handle REMOVE_USER_FROM_COLLECTION_FAILURE 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeUserFromCollection it should handle REMOVE_USER_FROM_COLLECTION_REQUEST 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;

exports[`CollectionReducer removeUserFromCollection it should handle REMOVE_USER_FROM_COLLECTION_SUCCESS 1`] = `
Object {
  "alertMessage": "",
  "collectionPrompt": false,
  "createDate": "",
  "isLocked": false,
  "items": Array [],
  "name": "",
  "openAlert": false,
  "users": Array [],
}
`;
